src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault/Customized/Authentication/KeyVaultCredential.cs

```diff
@@ -78,6 +89,20 @@ public KeyVaultCredential(KeyVaultClient.AuthenticationCallback authenticationCa
             return null;
         }
 
+        public override void InitializeServiceClient<T>(ServiceClient<T> client)
```

Comment:
Who calls this, and when?  It can't be called until the KeyVaultClient is created, but to create the KeyVaultClient you need a KeyVaultCredential.

Reply:
This is called in the generated code of the KeyVaultClient, and is called from all constructors which take credentials. 

https://github.com/Azure/azure-sdk-for-net/blob/0fa01adb3a1ef8fa13940c648a76ce439c72e8b4/src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault/Generated/KeyVaultClient.cs#L112-L123
https://github.com/Azure/azure-sdk-for-net/blob/0fa01adb3a1ef8fa13940c648a76ce439c72e8b4/src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault/Generated/KeyVaultClient.cs#L112-L123

All of the constructors in the customized KeyVaultClient partial class in turn call through the one of this generated constructors.  So it will always be called when the KeyVaultClient is instantiated.

https://github.com/Azure/azure-sdk-for-net/blob/0fa01adb3a1ef8fa13940c648a76ce439c72e8b4/src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault/Customized/KeyVaultClient.cs#L41-L42
https://github.com/Azure/azure-sdk-for-net/blob/0fa01adb3a1ef8fa13940c648a76ce439c72e8b4/src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault/Customized/KeyVaultClient.cs#L51-L52
https://github.com/Azure/azure-sdk-for-net/blob/0fa01adb3a1ef8fa13940c648a76ce439c72e8b4/src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault/Customized/KeyVaultClient.cs#L61-L62

Reply:
@schaabs Got it.  Thanks.